### PR TITLE
feature/add anchor id to title component

### PR DESCRIPTION
**Pull request by szymontodads**
# Note:
Please review internal DS task DSPL-127 before reviewing if possible.

## Description
`<header class="hl-title ${classes @ join=' '}">
        ...
        <h2 id="${anchorId}"`
^ anchorId is configurable so we can use anchor links

JS integrates with it by use of https://developer.mozilla.org/en-US/docs/Web/CSS/:target and scrolls window to make sure that certain navigation elements do not cover it.

## Motivation and Context
As a content developer, I want to add a link to some part of a page so visitors would be moved to the exact section.
DSPL-127

## Screenshots (if appropriate)
Header is opacity 0.5 here for sake of screenshot to let you see how it was covering the Title Component

before - desktop:
before-desktop

after - desktop:
after-desktop

before - mobile:
before-mobile

after - mobile:
after-mobile

## Upgrade notes (if appropriate)
n/a

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) labeled with `bug`
- [x] New feature (non-breaking change which adds functionality) labeled with `enhancement`
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code standards](/CONTRIBUTING.md) of this project.
^ AFAIK
- [x] My change requires updating the documentation. I have updated the documentation accordingly.
^ component has sufficient description for Content Authors, JS class is documented with JSDoc

### CODE REVIEW

**Comment by szymontodads** [comment]
I have read the Dynamic Solutions Individual Contributor License Agreement (CLA) Document and I hereby sign the CLA

File: core/src/main/java/pl/ds/howlite/components/models/TitleComponent.java [hunk 1]
```diff
@@ -55,6 +55,11 @@ public class TitleComponent implements Styled, Grid {
   @Default(values = "Add your text here")
   private String subtitle;
 
+  @Inject
+  @Getter
+  @Default(values = "Add your anchor-id here")
```
Comment (paweljozwicki) [inline core/src/main/java/pl/ds/howlite/components/models/TitleComponent.java:60]: For sure we cannot use a default value here. In that case, most of titles will have a "Add your anchor-id here" id.

File: core/src/main/resources/libs/howlite/components/title/dialog/.content.json [hunk 2]
```diff
@@ -20,6 +20,12 @@
         "label": "Heading text",
         "required": true
       },
+      "anchorId": {
+        "sling:resourceType": "wcm/dialogs/components/textfield",
+        "name": "anchorId",
+        "default": "Add your #anchorId here",
```
Comment (paweljozwicki) [inline core/src/main/resources/libs/howlite/components/title/dialog/.content.json:26]: I think we have no "default" property defined for textfield? But even if I'm wrong we should not define the default value for this field.

File: core/src/main/java/pl/ds/howlite/components/models/TitleComponent.java (hunk 1 above)
Comment (paweljozwicki) [reply core/src/main/java/pl/ds/howlite/components/models/TitleComponent.java:60]: You can use a `@Model(adaptables = Resource.class, defaultInjectionStrategy = OPTIONAL)` as a class annotation

**Comment by Rafal-DS** [comment]
I have read the Dynamic Solutions Individual Contributor License Agreement (CLA) Document and I hereby sign the CLA

File: core/src/main/resources/libs/howlite/components/title/dialog/.content.json (hunk 2 above)
Comment (Rafal-DS) [reply core/src/main/resources/libs/howlite/components/title/dialog/.content.json:26]: fixed

File: core/src/main/java/pl/ds/howlite/components/models/TitleComponent.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
 import pl.ds.howlite.components.Grid;
 import pl.ds.howlite.components.Styled;
 
-@Model(adaptables = Resource.class)
+@Model(adaptables = Resource.class, defaultInjectionStrategy = OPTIONAL)
```
Comment (paweljozwicki) [inline core/src/main/java/pl/ds/howlite/components/models/TitleComponent.java:30]: I think that
import static org.apache.sling.models.annotations.DefaultInjectionStrategy.OPTIONAL;
is missing :)

Comment (Rafal-DS) [reply core/src/main/java/pl/ds/howlite/components/models/TitleComponent.java:32]: Sorry about that :) added

**Review by paweljozwicki — APPROVED** [review]

**Pull request merged**